src/gui/util_gui.cpp

```diff
@@ -378,7 +380,26 @@ namespace Util {
 		return maybe_markup;
 	}
 
-	
+
+	namespace Tools
+	{
+		/**
+		 * Returns a new QToolButton with a unified appearance.
+		 */
+		QToolButton* newToolButton(const QIcon& icon, const QString& text, const char* whats_this)
```

Comment:
From the context in this file, I think the expected code pattern is more like:

~~~
namespace ToolButton
{
    QToolButton* create(const QIcon& icon, const QString& text, const char* whats_this)
    {
...
~~~

But I admit that this would require more changes add the calling spots, changing them to:
~~~
    auto* button = Util::ToolButton::create(...)
~~~
unless wrapping this call into small inline definitions of `newToolButton(()`.

Reply:
The idea behind the `Tools` name that I will also move the duplicated toolbar setup code into this namespace one day. Then `Tools::newToolButton()` and `Tools::newToolBar()` would be the symbols in the namespace. But we are not yet there, so my name choice is a typical YAGNI instance.

Your proposal sounds much better to me, even at the cost of changing the call sites. So, if `Util::ToolButton::create()` is your final proposal, I'll implement it.

Reply:
I'm not planning to invent more proposals, so the existing pattern is a "final proposal". But it is really just a proposal for consideration. If you don't want to adopt it now, you might also a add `\todo` comment. Or just to decide to leave your code as is.